includes/admin/class-sensei-course-pre-publish-panel.php

```diff
@@ -77,6 +79,33 @@ public function maybe_publish_lessons( $course_id ) {
 			return;
 		}
 
+		$publishing_meta_key = '_sensei_course_publishing_started';
+
+		// Even if the course is already published, each subsequent updates also triggers this hook anyway
+		// which caused the bug https://github.com/Automattic/sensei/issues/7555.
+		// So we need to check if it's an actual publish call.
+		$is_main_publish_call = 'publish' !== $old_status;
+
+		if ( $is_main_publish_call ) {
+			// This is the first call made, it's not the structure saving call, so the added/updated lessons are not yet saved at this point.
+			// So we set the flag to publish lessons on the next call, which is made after the structure is saved.
+			update_post_meta( $course_id, $publishing_meta_key, true );
+		}
+
+		$is_publishing_started = get_post_meta( $course_id, $publishing_meta_key, true );
+
+		if ( ! $is_main_publish_call && ! $is_publishing_started ) {
+			// If its not the "Publish" call and the flag is not set, then we don't need to publish lessons.
+			// Because it that case it's just a normal "Update" call.
+			return;
+		}
+
+		if ( ! $is_main_publish_call ) {
```

Comment:
If we're able to return as per my earlier comment, then I believe we may not need this `if` condition, because the only other possibility is that this is _not_ the main publish call and publishing _has_ started.

Reply:
Thanks! I've update all the comments here https://github.com/Automattic/sensei/pull/7582/commits/2ac96e8a96805b8b7c3a4cd46db0f518e4569c06